### ISSUE TITLE
How to improve effects of `pixi global install mercurial` ?

### ISSUE THREAD

**Issue by paugier** [prefix-dev/pixi#2688]
## Background

Mercurial is a Python application using C and Rust extensions. It is extendable with Mercurial extensions and two Python packages provide very useful Mercurial extensions that nearly all Mercurial users should use : hg-git (Mercurial extension `hggit`) and hg-evolve (Mercurial extensions `topic` and `evolve`).

These things are packaged in 3 conda-forge packages: mercurial, hg-git, hg-evolve.

To use Mercurial extensions, one has to write few lines in a configuration file (`~/.hgrc`).

I worked on a simple Python tool called `conda-app` (https://github.com/fluiddyn/conda-app) to install Mercurial globally with important extensions enabled. conda-app is a bit like pipx but for conda-forge. It simplifies the life of beginners and works well on Unix and Windows. With Miniforge, one has to do something like:

```sh
pipx install conda-app
# or
# conda activate base
# pip install conda-app
conda-app install mercurial
```

`conda-app` uses `mamba` or `conda` (which have to be installed) to install the right packages in an environment and setup Mercurial (writes a minimal and correct `~/.hgrc`). In practice, `conda-app` checks if a conda-forge package `mercurial-app` exists, finds it (because it exists) and installs it. mercurial-app depends on mercurial, hg-git and hg-evolve. I did that because the conda-forge mercurial package does not depend on Mercurial extensions (it would not make sense because these are the Mercurial extensions that depend on Mercurial).

## My question regarding pixi

It would be nice if `pixi global install mercurial` could do a little bit more than installing https://prefix.dev/channels/conda-forge/packages/mercurial globally. The post install call (setting ~/.hgrc) could be handle separately, but it would be nice that it actually installs https://prefix.dev/channels/conda-forge/packages/mercurial-app globally instead of https://prefix.dev/channels/conda-forge/packages/mercurial.

I don't think there are any mechanism to be able to do that with conda-forge and Pixi.

**Comment by Hofer-Julian** [comment]
You are right, optional dependencies are not yet a thing, but @wolfv is working on it 😉 
https://gist.github.com/wolfv/9a84b8881826851de6ff44e1e88fd4ad

Looking at [uv tool install](https://docs.astral.sh/uv/reference/cli/#uv-tool-install) I don't see how they utilize optional dependencies. Am I missing something?

**Comment by paugier** [comment]
> Looking at [uv tool install](https://docs.astral.sh/uv/reference/cli/#uv-tool-install) I don't see how they utilize optional dependencies.

Simply
```
uv tool install black[jupyter]
```

**Comment by Hofer-Julian** [comment]
Until optional dependencies are a thing, you can add packages explicitly with:

 ```
pixi global install mercurial --with hg-git --with hg-evolve
```

**Comment by Hofer-Julian** [comment]
I would say so, there isn't much we can do to improve the experience at the moment
